Python/Algorithms/linked_list_structs.py

```diff
@@ -0,0 +1,185 @@
+
+
+class Node:
+    def __init__(self, val, next = None):
+        """
+        Creates a linked list node with given value `val` and points to `next` node.
+
+        params:
+            val (Any): Value of the node
+            next (Node): Next node in the linked list
+        """
+        self.value = val
+        self.next = next
+
+
+class FIFO:
+    def __init__(self, value=None):
+        """
+        Starts the FIFO structure with given value
+
+        params:
+            value (Any): value of the first element of the FIFO structure
+        """
+
+        self.head = Node(value) if value else None
+        self.last = self.head
+        self.length = 1 if self.head else 0
```

Comment:
I didn't understand. Should I implement linked lists and then overload the methods from the linked list? That'll be an overhaul. I implemented just the Linked List nodes which have two attributes `value` and `next`. Using those nodes I'm creating a linked list within the FIFO structure.

Reply:
No That will never be overhaul...you just have to choose where you want to add it...for stack you would add it at the head and for queue, you would add it at the tail simple

Reply:
The approach is the same right now as well. I'll add another `LinkedList` class.
So the new heirarchy will be.
class `LinkedList` will have `insert`, `remove`, `isEmpty` and `size` methods. FIFO and LIFO will inherit those methods. `FIFO` will not really change anything with the methods. `LIFO` will overload `insert` method to add elements to the head instead of the tail.
Is that approach correct?